Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselPanel.cs

```diff
@@ -0,0 +1,182 @@
+        }
+
+        private static void OnOrientationChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            (d as CarouselPanel).InvalidateArrange();
+            var carousel = d as CarouselPanel;
+            carousel.ManipulationMode = (Orientation)e.NewValue == Orientation.Vertical ?
+                Windows.UI.Xaml.Input.ManipulationModes.TranslateY : Windows.UI.Xaml.Input.ManipulationModes.TranslateX;
+        }
+
+        protected override Size ArrangeOverride(Size finalSize)
+        {
+            double centerLeft = finalSize.Width / 2;
+            double centerTop = finalSize.Height / 2;
+
+            this.Clip = new RectangleGeometry { Rect = new Rect(0, 0, finalSize.Width, finalSize.Height) };
+
+            for (int i = 0; i < this.Children.Count; ++i)
+            {
+                var child = this.Children[i];
+                double deltaFromSelectedItem = 0;
+                if (i < ItemIndex)
+                {
+                    // item is above
+                    for (int j = i; j < this.ItemIndex; ++j)
+                    {
+                        var value = Orientation == Orientation.Vertical ? Children[j].DesiredSize.Height : Children[j].DesiredSize.Width;
+                        deltaFromSelectedItem -= value + ItemSpacing;
+                    }
+                }
+                else if (i > ItemIndex)
+                {
+                    for (int j = this.ItemIndex; j < i; ++j)
+                    {
+                        var value = Orientation == Orientation.Vertical ? Children[j].DesiredSize.Height : Children[j].DesiredSize.Width;
+                        deltaFromSelectedItem += value + ItemSpacing;
+                    }
+                }
+
+                double childLeft, childTop;
+                if (Orientation == Orientation.Vertical)
+                {
+                    childLeft = centerLeft - (child.DesiredSize.Width / 2);
+                    childTop = centerTop - (Children[ItemIndex].DesiredSize.Height / 2) + deltaFromSelectedItem;
+                }
+                else
+                {
+                    childLeft = centerLeft - (Children[ItemIndex].DesiredSize.Width / 2) + deltaFromSelectedItem;
+                    childTop = centerTop - (child.DesiredSize.Height / 2);
+                }
+
+                child.Arrange(new Rect(0, 0, child.DesiredSize.Width, child.DesiredSize.Height));
+
+                var visual = ElementCompositionPreview.GetElementVisual(child);
+                var offsetAnimation = visual.Compositor.CreateVector3KeyFrameAnimation();
+                offsetAnimation.Duration = TimeSpan.FromMilliseconds(200);
+                offsetAnimation.InsertKeyFrame(1, new System.Numerics.Vector3((float)childLeft, (float)childTop, 0));
+                visual.StartAnimation("Offset", offsetAnimation);
+
+                (child as CarouselItem).CarouselItemLocation = i - ItemIndex;
```

Comment:
Should we exchange the location of both ? because in CarsouelItemLocation you're invoking an event and at this point of time IsCenterd is not updated yet with the new information so if you used IsCentered in CarouselItem_CarouselItemLocationChanged isCentered will have the old value not the updated one because of the order of execution (Also if you exchange the 2 lines you'll be able to use IsCentered instead of item.CarouselItemLocation == 0 )
Will point to it into another comment on the sample page
(child as CarouselItem).IsCentered = i == ItemIndex;
(child as CarouselItem).CarouselItemLocation = i - ItemIndex;